spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java

```diff
@@ -153,10 +153,15 @@ public ArrowColumnVector getChild(int ordinal) {
     return accessor.childColumn(ordinal);
   }
 
-  static ColumnVector forHolder(VectorHolder holder, int numRows) {
-    return holder.isDummy() ?
-        new ConstantColumnVector(Types.IntegerType.get(), numRows, ((ConstantVectorHolder) holder).getConstant()) :
-        new IcebergArrowColumnVector(holder);
+  static ColumnVector forHolder(VectorHolder holder, int numRows, boolean[] isDeleted) {
+    if (holder.isDummy()) {
+      if (holder instanceof VectorHolder.IsDeletedVectorHolder) {
+        return new DeletedMetaColumnVector(Types.BooleanType.get(), isDeleted);
+      }
+      return new ConstantColumnVector(Types.IntegerType.get(), numRows, ((ConstantVectorHolder) holder).getConstant());
```

Comment:
should we also check `instanceOf` ConstantVectorHolder in else-if, then for the unknown type we throw an exception?

Reply:
It is safer that way. Although it throws anyway if the casting doesn't work.
```
(ConstantVectorHolder) holder
```

Reply:
Checked and refactored this part of code by adding a column vector builder.